packages/babel-plugin/src/dom__experimental.tsx

```diff
@@ -0,0 +1,114 @@
+import { declare } from '@babel/helper-plugin-utils';
+import * as t from '@babel/types';
+import { transformCss } from '@compiled/css';
+import { kebabCase, unique } from '@compiled/utils';
+
+import { buildCodeFrameError } from './utils/ast';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const PKG_JSON = require('../package.json');
+const COMPILED_MODULE_EXPERIMENT = '@compiled/dom__experimental';
+const COMPILED_INSERT_STYLES_CALL = 'insertStyles';
+const COMPILED_STYLE_CALL = ['cstyle', 'create'];
+
+const objToCss = (obj: Record<string, any>): string => {
+  let css = '';
+
+  for (const key in obj) {
+    const val = obj[key];
+    if (typeof val === 'object') {
+      const value = objToCss(val);
+      css += `${key} { ${value} }`;
+    } else {
+      css += `${kebabCase(key)}: ${val};`;
+    }
+  }
+
+  return css;
+};
```

Comment:
Why not use `[postcss-js](https://github.com/postcss/postcss-js)` here to prevent edge cases?